src/NHibernate.Test/SqlTest/Query/NativeSQLQueriesFixture.cs

```diff
@@ -294,23 +276,64 @@ void AssertQuery(bool fromCache)
 							.List();
 					t.Commit();
 
-					Assert.AreEqual(1, l.Count);
-					//TODO: Uncomment if we properly fix caching auto discovery type queries with transformers
-					// var msg = "results are expected from " + (fromCache ? "cache" : "DB");
-					// Assert.That(Sfi.Statistics.QueryCacheMissCount, Is.EqualTo(fromCache ? 0 : 1), msg);
-					// Assert.That(Sfi.Statistics.QueryCacheHitCount, Is.EqualTo(fromCache ? 1 : 0), msg);
+					Assert.That(l.Count, Is.EqualTo(1));
```

Comment:
Can you also add similar test checking caching for empty list? (basically copy/paste removing  test data creation part and adjusting this check)

Reply:
Done, with multiple fixes.